### DISCUSSION TITLE
Memory leak when AllowCollisionTesting is set to false after a collision has been found.

### DISCUSSION THREAD

**Discussion by ghost** [bepu/bepuphysics2#215] [Q&A] [answered]
I'm using BepuPhysics for collision checks only. The implementation is based on the CollisionQueryDemo. When I check a compound with a mesh, and a collision has been found, I want to stop the check and report a collision back to the system. I'm doing this by setting AllowCollisionTesting to false when a collision has been found. When I'm setting AllowCollisionTesting to false, the memory usage ramp up from 500mb to, 7000mb. Is there another way to stop the collision batcher after a collision has been found, or a way to fix memory leaking.

**Comment by RossNordby** [comment, marked as answer]
That shouldn't happen- I've reproduced the bug, working on a fix.

**Reply by RossNordby** [reply]
Should be fixed in latest; 2.5.0-beta.2 package should be up when CI finishes.

Thanks for the report!

**Reply by ghost** [reply]
Thank you for the quick fix. I'm working on the 2.3.1 branch and implement the fix for this branch and it works. Can I create a pull request for this fix on 2.3.1 or does this fix need further testing?

**Reply by RossNordby** [reply]
Sure, a PR would be great! The fix has passed some pretty arduous fuzztesting, so it *should* be good to go.
